modular_skyrat/modules/wargame_projectors/code/projectors.dm

```diff
@@ -0,0 +1,193 @@
+/obj/item/wargame_projector
+	name = "holographic projector"
+	desc = "A handy-dandy holographic projector developed by Nanotrasen Naval Command for playing wargames with, this one seems broken."
+	icon = 'modular_skyrat/modules/wargame_projectors/icons/projectors_and_holograms.dmi'
+	icon_state = "projector"
+	inhand_icon_state = "electronic"
+	worn_icon_state = "electronic"
+	lefthand_file = 'icons/mob/inhands/items/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items/devices_righthand.dmi'
+	force = 0
+	w_class = WEIGHT_CLASS_SMALL
+	throwforce = 0
+	throw_speed = 3
+	throw_range = 7
+	item_flags = NOBLUDGEON
+	greyscale_config = /datum/greyscale_config/wargame_hologram_projector
+	greyscale_colors = COLOR_WHITE
+	/// All of the signs this projector is maintaining
+	var/list/projections
+	/// The maximum number of projections this can support
+	var/max_signs = 1
+	/// The color to give holograms when created
+	var/holosign_color = COLOR_WHITE
+	/// The type of hologram to spawn on click
+	var/holosign_type = /obj/structure/wargame_hologram
+	/// A list containing all of the possible holosigns this can choose from
+	var/list/holosign_options = list(
+		/obj/structure/wargame_hologram,
+	)
+	/// Contains all of the colors that the holograms can be changed to spawn as
+	var/static/list/color_options = list(
+		"Red" = COLOR_RED_LIGHT,
+		"Orange" = COLOR_LIGHT_ORANGE,
+		"Yellow" = COLOR_VIVID_YELLOW,
+		"Green" = COLOR_VIBRANT_LIME,
+		"Blue" = COLOR_BLUE_LIGHT,
+		"Pink" = COLOR_FADED_PINK,
+		"White" = COLOR_WHITE,
+		"Gray" = COLOR_GRAY,
+		"Brown" = COLOR_BROWN,
+		"Ice" = COLOR_BLUE_GRAY,
+	)
+
+/obj/item/wargame_projector/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/openspace_item_click_handler)
+	set_greyscale(holosign_color)
+
+/obj/item/wargame_projector/handle_openspace_click(turf/target, mob/user, proximity_flag, click_parameters)
+	afterattack(target, user, proximity_flag, click_parameters)
+
+/obj/item/wargame_projector/examine(mob/user)
+	. = ..()
+	if(projections)
+		. += span_notice("It is currently maintaining <b>[projections.len]/[max_signs]</b> projections.")
+	. += span_notice("Use the projector <b>in hand</b> to change what type of hologram it creates.")
+	. += span_notice("<b>Alt clicking</b> the projector will let you change the color of the next hologram it makes.")
+	. += span_warning("<b>Control clicking</b> the projector will allow you to clear all active holograms.")
+
+/// Changes the selected hologram to one of the options from the hologram list
+/obj/item/wargame_projector/proc/select_hologram(mob/user)
+	var/list/choices = list()
```

Comment:
i feel like these could be cached on the projector so it only has to be generated once instead of every time one is selected